### ISSUE TITLE
[QA] Automation test for Desktop Client 2.9

### ISSUE THREAD

**Issue by swoichha** [owncloud/client#8971]
### Automation test for Desktop Client

[Test Plan Template](https://github.com/owncloud/client/issues/8870)  This issue has a list of Testplan Release Desktop 2.9

### Background
While doing manual testing for Desktop 2.9 we came across many tests that can be automated. In this issue are the list of test that can and should be automated.

 ----
* [x]  1. Install/Update - Connect - Login @swoichha 
-> https://github.com/owncloud/client/issues/8971#issuecomment-909884138
* [x] 2. Folders @Talank
->https://github.com/owncloud/client/issues/8971#issuecomment-909884257
* [ ] 3. Files   @Talank
-> https://github.com/owncloud/client/issues/8971#issuecomment-909884348
* [x] 4. ~~Move files and folders~~  @swoichha 
-> https://github.com/owncloud/client/issues/8971#issuecomment-909884563
* [x] 5. Edit Files @swoichha 
-> https://github.com/owncloud/client/issues/8971#issuecomment-909884684
* [x] 6. Delete Files and Folders  @swoichha 
-> https://github.com/owncloud/client/issues/8971#issuecomment-909884844
* [x] 7. Sync process  
-> https://github.com/owncloud/client/issues/8971#issuecomment-909884918
* [x] 8. ~~Without connection~~ @swoichha 
-> https://github.com/owncloud/client/issues/8971#issuecomment-909884992
* [x] 9. Sharing 
* [x] 9.1 Share link @swoichha 
-> https://github.com/owncloud/client/issues/8971#issuecomment-909885070
* [x] 9.2 Share with Users and Groups @saw-jan
-> https://github.com/owncloud/client/issues/8971#issuecomment-909885166
* [x] 10. Selective_Sync @saw-jan 
* [x] 10.1 Choose what to sync from the setup window 
-> https://github.com/owncloud/client/issues/8971#issuecomment-909885292
* [x] 10.2 "Skip folders configuration"  
* [x] 10.3 "Add Folder..." @swoichha 
-> https://github.com/owncloud/client/issues/8971#issuecomment-909885541

**Comment by swoichha** [comment]
### 1. Install/Update - Connect - Login
Existing squish tests concerning this topic: tst_addAccount (also tests multiple accounts), tst_loginLogout, tst_removeAccountConnection

* For this test set, choose a random platform win/lin/mac, unless test items specify otherwise.
* Record which platform was chosen: ...

TestID | Test Case | Steps to reproduce |Expected Result |Result| Related Comment
------------ | ------------- | ------------- | ------- | ----- | ------
1 | Update Installation | 1. You need to have installed a previous version 2. Update to the new version |Test on Win, macOS, Linux| :heavy_check_mark:  Win,<br> :heavy_check_mark:  macOS,<br> :heavy_check_mark: Linux  | |
2 | Install the new version | 1. Delete the previous version 2. Install the new version || :heavy_check_mark:  Win,<br> :heavy_check_mark:  macOS,<br> :heavy_check_mark: Linux | |
3 | Verify that you can enter a server address (valid server cert)| 1. Launch desktop app 2. enter a server address 3. Click on Next 4. The lock is shown closed - SSL details on mouse over| | :heavy_check_mark: | |
4 | Verify that you can enter a server address (self signed cert)| 1. Launch desktop app 2. enter a server address 3. Click on Next 4. If it is the first time you should accept the certificate - click on the checkbox and OK (The lock is shown in closed - the server validated SSL connection window is opened)| | :heavy_check_mark: | |
5 :robot: extend tst_addAccount for wrong credentials| Introduce username/password  | 1. Fill in username and password |If the credentials are not correct a message is shown "Error: Wrong credentials" | :robot: tst_addAccount (except wrong credentials) :heavy_check_mark: |  |
6 | Verify that all contents of the server account is synced to the local folder | 1. Choose to sync everything from server (default option) 2. Select the local folder desired | All the files/folders are synced down |:robot: tst_addAccount :heavy_check_mark:  | |
7 | Verify that only the folder(s) selected are synced in the local folder  | 0. Disable VFS, 1. Click on Choose what to sync 2. The remote folder(s) are shown, select which you want to sync 3. Select the local folder |The selected folders selected are synced | :heavy_check_mark: | |
8 | Verify that you can skip folder configuration | 1. Click on "Manually create folder sync connection" | No settings dialog opens, the account shows only an "Add Folder Sync Connection" button.  | :heavy_check_mark: |   |
9 | Connect to a server with LDAP (or optinally AD) | 0. Only for minor releases, 1. Setup owncloud server with openldap, 2. Connect. | Make sure no technical user name shows up in UI (e.g. account name, sharing...) | :heavy_check_mark: | uuid and ldap-account-name can be used to initiate sharing, but these never show up in the UI |
10 | Connect to a 8.1.x server | 0. Only for minor releases, 1. `docker run -ti -p 8181:80 owncloud:8.1`, 2. Connect.  A warning appears: "The server version 8.1.12.2 is unsupported", but syncing works. | :heavy_check_mark: | |

**Comment by swoichha** [comment]
### 2. Folders
* For this test set, choose a random platform win/lin/mac, unless test items specify otherwise.
* Record which platform was chosen: ...
- 'Go to local sync folder and create a folder' means: "Open folder" on client dot ... menu, create a new folder in file browser
- 'Verify on the server' means:  the folder is sent from client to server / check either with a) web browser, b) another client, or c) via server introspection

TestID | Test Case | Steps to reproduce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 | Verify that you can create one folder  | 1. Go to local sync folder 2. Create a single folder 3. Wait for sync | The folder is visible on the server | :robot: tst_syncing :heavy_check_mark: | |
2 :robot: can be added| Verify that you can create one folder with long name  | 1. Go to local sync folder 2. Create a single folder with a long name (59 characters+a terminating zero bytes) 3. Wait for sync |The folder visible on the server | :heavy_check_mark: | |
3 :robot: can be added| Verify that you can create one folder with special characters in the name | 1. Go to local sync folder 2. Create a single folder with a special character(s) in the name (e.g $%ñ&) 3. Wait for sync | The folder appears on the server | :heavy_check_mark:  | |
6 :robot: can be added| Verify that you can sync many subfolders | 1. Go to local sync folder 2. Copy a folder with 5 empty subfolders and 5 folders containing files 3. Wait for sync | All 10 subfolders are visible on the server | :heavy_check_mark:  | |
8 | Verify that you can create multiples folders at once | 1. Create a folder with 400 subfolders outside the sync root, 2. Move that into the sync root, 3. Wait for sync  | All 400 folders appear on the server | :heavy_check_mark: | |
9 :robot: can be added| Verify that you can copy one folder | 1. Go to local sync folder 2. Create a single folder with some files in it 3. Copy and paste the folder 4. Wait for sync | Both copies appears on the server | :heavy_check_mark: | |
10 :robot: can be added| Verify that you can create a subfolder with long name | 1. Go to local sync folder, 2. Create a folder called "Folder1" 3. Create a subfolder called "LUsgzq!0k02sek+szBqrzN5=R#UJpWql&rwhnYVb~Gh!l!”  (optionally up to 255 characters), 4. This subfolder had a file called ilppng.PNG inside it 5. Wait for sync |The files are synced correctly| :heavy_check_mark: | |
11| Verify pre existing folders in local (Desktop client) are copied over to the server | 1. Turn off the Desktop client 2. Go to local sync folder 3. Create several folders inside the Desktop Client folder at several different levels, 4. Turn ON the Desktop Client  | Folders appear on the server | :heavy_check_mark: | |
12| Filenames that are rejected by the server are reported | 1. Go to local sync folder, 2. Create a file called `"a\\a"` (or another name not accepted by the server), 3. Wait for sync | The sync status reports an error, the file s not synced| :heavy_check_mark:   |  |
-- | **reviewed up to here** | -- ||
13| Verify one empty folder with a length longer than the allowed limit will not be synced | 1. Go to local sync folder 2. Create a single empty folder with a name longer than that allowed by ILP (more than 59 characters) 3. Look at the Via Web repository 4. Repeat this with a folder at the root level, and in various subfolders up to 5 levels deep 5. Sync |At the Via Web the folder has not been synced| :heavy_check_mark: | |
14| Sync works for .zip/.rar files with elaborate internal folder structures | 1. Create a .zip file with many internal folders and files 2. Copy the .zip file to the Desktop Client folder 3. Unzip the .zip file inside the Destop Client folder |1. Make sure you get a popup saying that all the extracted files have synced. 2. Look at Via Web and make sure that the folder has been synced over| :heavy_check_mark: :x:  | Sync works - Popup Message don't came up |
15| Files that error with API should try 3 times, and then blacklist | 1. Try to sync a folder that has more than 65 characters 2. Then sync it with some contents, it should try three times and then be blacklisted 3. If you rename the folder it should try again, and succeed if the name is less than 65 characters |The folder is synced| :heavy_check_mark: | |
16| Invalid system names | 1. On the server, create folders named 'COM' and 'test%' and two files named 'PRN' and 'foo%' | A MacOS client syncs down 'COM' and 'PRN' but not 'test% or 'foo%' | :construction: | |
17| Invalid system names | 1. On the server, create folders named 'COM' and 'test%' and two files named 'PRN' and 'foo%' | A windows client syncs down 'test%' and 'foo%' but not 'COM' or 'PRN' | :heavy_check_mark:  | Folder named COM will be synced. |
18| Invalid system names | 1. On the server, create folders named 'COM' and 'test%' and two files named 'PRN' and 'foo%' | A Linux client syncs down all. | :heavy_check_mark:  | All folder will be synced |

**Comment by swoichha** [comment]
### 3. Files

TestID | Test Case | Steps to reproduce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 | Via User should see the single file on Via Web when the file is successfully sync in Desktop Client sync folder | 1. Drop one selected file in Desktop Client sync folder 2. User can see end receive notification from System tray that single file is successfully added in Desktop Client sync folder 3. Go to Desktop Client system tray icon and select launch Via Web| Verify that user can see that the single file is sync on Desktop Clien Sync folder and is available on Via Web| :robot: tst_syncing What about systray notifications? :heavy_check_mark: | File is synced. But no Popup Notification will be shown |
2 | Via user should see the multiple file on Via Web when the files or folders are successfully sync in Desktop Client Sync folder | 1. Drop multiple selected files in Desktop Client Sync folder 2. User can see and receive notification from System tray that multiple files or folders are successfully added in Desktop Client Sync folder 3. Go to Desktop Client System tray icon and select launch Via Web|Verify that User can see that multiple files are sync on Desktop Client folder and are available on Via Web| :heavy_check_mark: | File is synced. But no Popup Notification will be shown |
3| User adds file to the sync folder will show up in repository  | 1. Users see a completed icon overlay on the copied file(s) in the Desktop client folder 2. The Sync Files tab shows the files added in the Desktop Client folder from desktop|| :heavy_check_mark:  | |
4| User adds the various types of files | 1. Microsoft word documents, Microsoft Excel, Microsoft Powerpoint, .JPG, .PDF, .MP3|The Sync files tab shoe all type of files added in th Desktop Client folder from desktop| :heavy_check_mark:  | All files snyced and shown in Activities Tab under synchronization protocol |
5| Long file name  | 1. Create a file with this name "dfkjsdfjksdkdfjsdfuidjfkdsjfksdjfks djfksdjfksdjfksdjfkdsjfks djfkdsjfkdsfjlsdkfjsdkjflksdjfk lsdjfksdjfkdsjfkldsjfkldsjfkdsjfksd jfksdjfklsdjfklsdjflksdjflksdjfklsdj fklsdjfksdjfksdjfksdjfksdjfksdfj skdfjksdjfksdjfksdjfksdjfksd.pdf"|Sync is successful| :heavy_check_mark: | |
6| User copies/drag&drops multiple files at a time to the Sync folder | 1. Users see the completed icon overlay on all type of files in Desktop Client folder on Desktop. 2. The Sync files tab shows all type of files added in the Desktop Client folder from Desktop|Sync is successful| :heavy_check_mark:  | |
7| Sync files at the same time| 1. Add a file of 1MB onthe local inside folder, and in the same time, add another file on the same remote folder (make sure that file is on the server before the client finishes)|Look at Via Web and the Desktop Client to make sure that the sync is correct| :x: | File will not synced by the client |
8| Same name files, different extension | 1. Create the same name file with two differents extensions|The files are sync corectly| :heavy_check_mark: 
 | |
9| Files with spaces in the name | 1. Move a file under the root sync folder with spaces in the name 2. Let them sync|The files are sync correctly| :heavy_check_mark: | |
10| Create and delete one file with special characters in the name| 1. Go to Desktop Client 2. Create a single folder 3. Move with in this folder a file with a special characters in the name fo example "~`!@#$^&()-_=+{[}];'," 4. Wait for the file to sync to the Via Web 5. Delete the file in Desktop Client |Look at Via Web and make sure that the file got deleted| :heavy_check_mark:  | |
11| User copies large files to the Desktop Client folder on the Desktop| 1. User see in-progress overlay icon while upload is in progress in Desktop 2. User sees the completed sync icon overlay once the upload process is done|The Sync files tab shows file added in th Desktop Client folder from desktop| :heavy_check_mark:  | |
12| upload 50Mb file| 1. Create a folder under sync folder 2. Copy a large file (50MB) in this folder|After several sync processes the file is correct| :heavy_check_mark: | |
13| Upload 150Mb files| 1. Upload a large .pdf (100-150MB) file through web browser|Sync is successful| :heavy_check_mark: | |
14| Upload 3000Mb files| 1. Upload a large .pdf (3GB) file through web browser|Sync is successful| :heavy_check_mark: | |
15| Upload 2048Mb files| 1. Upload a large file (2GB=2147483648 bytes) through web browser|Sync is successful| :heavy_check_mark:  | |
16| Upload 1000Mb files| 1. Upload a folder 1000 file (1Mb each) |Sync is successful| :heavy_check_mark: | |
17| Upload 500Mb+500Mb files| 1. Upload two folders with 500 files each |Sync is successful| :heavy_check_mark: | |
18| Upload 1024Mb file| 1. Upload a 1GB file |Sync is successful| :heavy_check_mark: | |
19| Verify the limit of the quota| 1. Upload the necessary large files to fill the quota |Warning: "The available space of your workspace is running out, please delete some files to free space"| :heavy_check_mark: :x: | A different Message will be shown |

**Comment by swoichha** [comment]
### 4. Move files and folders

TestID | Test Case | Steps to reprouce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 | Move from sublevels to root | 1. Move a couple of files (File1 and File2) with different content under the root sync folder 2. Let them sync 3. Move other file under the root sync folder| The content of files is correct | :heavy_check_mark: | |
2 | Move folder down | 1. Move one folder from sync root to a 5 deep level folder 2. Sync| The content of the folder is correct | :heavy_check_mark: | |
3 | Move folder up | 1. Move one folder from 5 deep level folder to the sync root 2. Sync| The content of the folder is correct | :heavy_check_mark:  | |
4 | Move files from one folder another| 1. Move a couple of files under the root sync folder 2. Create a folder 3. Let them sync 4. Move the files to the new folder 5. Let them sync| The files in the correct size in Via Web| :heavy_check_mark:  | |
5 | Move two or more folders down | 1. Move two or more folders from sync root to a 5 dep level folder 2. Sync| The content of the folder is correct | :heavy_check_mark: | |
6 | Move two or more folders up | 1. Move one folder from 5 deep level folder to the sync root 2. Sync| The content of the folder is correct | :heavy_check_mark:  | |

**Comment by swoichha** [comment]
### 5. Edit Files
- For this test, I used Windows 10 (Enterprise Evaluation) version 21H1 (OS build 19043.1165)

TestID | Test Case | Steps to reprouce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 | Edit a .txt file| 1. Create a .txt file in a sync folder 2. Edit some text 3. Wait for it to sync 4. Modify the .txt file and add more content 5. Wait for it to sync 6. Modify the .txt file and add more content 7. Wait for it to sync | The file at the server had the same content after the sync is completed| ✔️ | |
2 | Edit a .doc file| 1. Create a .doc file in a sync folder 2. Edit some text 3. Wait for it to sync 4. Modify the .doc file and add more content 5. Wait for it to sync 6. Modify the .doc file and add more content 7. Wait for it to sync | The file at the server had the same content after the sync is completed| ✔️ | |
3 | Edit a .xls file| 1. Create a .xls file in a sync folder 2. Edit some content 3. Wait for it to sync 4. Modify the .xls file and add more content 5. Wait for it to sync 6. Modify the .xls file and add more content 7. Wait for it to sync | The file at the server had the same content after the sync is completed| ✔️ | |
4 | Replace a .pdf file| 1. Create a .pdf file in a sync folder 2. Replace it with a different pdf (but same name) 3. Wait for it to sync 4. Modify the .pdf file and add more content 5. Wait for it to sync 6. Modify the .pdf file and add more content 7. Wait for it to sync | The file at the server had the same content after the sync is completed| ✔️ | |
5 | Edit a file while the folder is renaming | 1. You should had any kind of file already sync 2. Go to Desktop Client 3. Open the file and edit it 4. Go to the Via Web and rename the folder 5. Sync with the oc-worker 6. Do not refresh the browser at the server and download the file edited | The file at the server had the same content| ❌ | The file is not synced when I perform the following actions simultaneously: 1. Update the file in desktop 2. Rename the parent folder on the web ->https://github.com/owncloud/client/issues/8921  |

**Comment by swoichha** [comment]
### 6. Delete Files and Folders

TestID | Test Case | Steps to reprouce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 | Delete one file| 1. Go to Desktop Client 2. Create or copy a single file with a short name (less than 20 characters) 3. Wait for the file to sync to the Via Web 4. Delete the file in Desktop Client|The file is been deleted on Via Web| :heavy_check_mark: | |
2 | Delete one file with long name| 1. Go to Desktop Client 2. Create or copy a single file with a long name (more than 240 characters) 3. Wait for the file to sync to the Via Web 4. Delete the file in Desktop Client|The folder is been deleted on Via Web| :heavy_check_mark: :x: | Maximum length of filename is 228 chraracter |
3 | Delete one folder| 1. Go to Desktop Client 2. Create a single folder with a short name (more than 20 characters) 3. Wait for the folder to sync to the Via Web 4. Delete the folder in Desktop Client|The folder is been deleted on Via Web| :heavy_check_mark: | |
4 | Delete one folder with long name| 1. Go to Desktop Client 2. Create a single folder with a long name (59 characters+a terminating zero bytes) 3. Wait for the folder to sync to the Via Web 4. Delete the folder in Desktop Client|The folder is been deleted on Via Web| :heavy_check_mark: | |
5 | Delete multiple files| 1. Go to Desktop Client 2. Create or copy a multiple files 3. Wait for the files to sync to the Via Web 4. Delete the file in Desktop Client|The files get deleted on Via Web| :heavy_check_mark: | |
6 | Delete large file (2048Mb)| 1. Go to Desktop Client 2. Create or copy a single big file (2GB) 3. Wait for the file to sync to th Via Web 4. Delete the file in Desktop Client|The file get deleted on Via Web| :heavy_check_mark:  | |

**Comment by swoichha** [comment]
### 7. Sync process

TestID | Test Case | Steps to reprouce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 | Move some files while sync| 1. Move a couple of files (File1 and File2) with different content under the root sync folder 2. Let them sync 3. Move other file under the root sync folder and while sync 4. Delete File1 and rename File2 with the name File1| The content of files is correct| :heavy_check_mark:  | |
2 | Rename the file while sync| 1. Create a .txt file with some text in it 2. Let it sync 3. Rename the file and let is sync 4. While #3 is sync, edit the content of the .txt file| The content of file is correct| :heavy_check_mark:  | |
3 | Deleted folders while sync| 1. Create a tree of folders+subfolders (e.g folder1, folder2, folder3 and some .txt files in this folder) 2. Delete the folders while sync| Look at Via Web and make sure that the folders got deleted| :heavy_check_mark: | |
4 | Deleted folders while sync with two clients| 1. Create a tree of folders+subfolders (e.g folder1, folder2, folder3 and some .txt files in this folder) 2. Delete some folders and keep another while sync 3. Sync with another client at the same time| Look at Via Web and make sure that the folders got deleted and the remains folders sync correctly| :heavy_check_mark:  | |
5 | Create a new folder with space ar the end| 1. Create a folder with space at end| Look if is sync| :question:  | If you create a folder with a space at the end, your os (nautilus @ ubuntu) will remove this space. The folder is syncing  |
6 | Deselect to the option "Allow apps to use the Share API"| 1. Install the new version 2. Go to Server Client 3. Sing in as Admin user 4. Go to Admin 5. On Sharing option, deselect to the option "Allow apps to use the Sahre API" 6. Go to the Desktop Client 7. Configure one account 8. Open folder 9. Create new folder (e.g Example1) 10. Right click on the folder Example1 11. Click on "Share with oC" 12. If you introduce the Password and press enter|| :question: | There is no option "share with oc". Sharing with disabled Shareing Api doesn't work. |
7 | If you share a folder with another user, and then rename the folder on the Desktop does not change the name| 1. Install the new version with 2 accounts (user1, user2) 2. Go to Server Client with user1 3. Create a new folder (e.g. Share with) 4. Click on Share, and share with user2 5. Go to  Desktop Client with user2 6. Wait sync 7. Go to Server client with user1 8. Rename the Folder (Share with) to (Share with user2" 9. Go to Desktop client with user2 11. Wait sync | The folder name has not change. The shared folder is actually a virtual mount point for all recipients, so renaming it doesn't affect all recipients. Only the contents of it will be visible for anyone. Works as expected| :heavy_check_mark:  | |

**Comment by swoichha** [comment]
### 8. Without connection


TestID | Test Case | Steps to reprouce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 | Upload several files and folder to the sync folder without internet| 1. Upload several files and folder without internet 2. The connection is back | The files and folder are sync with the server | :heavy_check_mark:  | |
2 | Upload several files and folder with special characters to the sync folder without internet| 1. Upload several files and folder without internet 2. The connection is back | The files and folder are sync with the server | :heavy_check_mark:  | |
3 | Upload the same folder from the server and form the client with differente files inside| 1. Upload several files and folder without internet 2. The connection is back | The files and folder are sync with the server | :heavy_check_mark:  | |
4 | Remove folder from the client| 1. Remove one folder from the local folder 2. The connection is back | The folder is not on the server | :heavy_check_mark: | |

**Comment by swoichha** [comment]
### 9. Sharing

#### 9.1 Share link

TestID | Test Case | Steps to reprouce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 | Share a Folder/File with special characters| 1. User share a folder (special characters /, <,>,:,?) with you; 2. On the server accept the share, (if needed);  3. Refresh the parent folder | File/Folder is not displayed as shared| :heavy_check_mark:  | |
2 | Share a Folder/File with special characters| 1. User share a folder (special characters in the name $%ñ&) with you 2. Refresh the parent folder | File/Folder is displayed as shared | :heavy_check_mark:  |  sharing works on Linux and window. TODO: try mac |
3 | Unshare the Folder/File | 1. User unshare a folder/file with you 2. Refresh the folder/file | File/Folder is not shown as shared| :heavy_check_mark:  | |
5 | Unshare the Folder with you | 1. at client navigate to shared folder with file browse or shell; 2. User unshares this folder with you 3. Upload file from Desktop Client | Folder error| :no_entry: :heavy_check_mark: | Win+Linux: No error. A new (unshared) folder gets created  |
6 | Verify shared icon | 1. User share a folder | When the folder/file are shared show a special icon | :heavy_check_mark:   | Need to enable two ownCloud plugins in Nemo |
7 | ~Verify shared icon~ | 1. User share a folder | When the folder or file is shared by link show a special icon | :construction: | ?duplicate? |
8 | Move a shared folder with permissions | 1. Move a shared folder with permissions inside another folder from Desktop; 2. The needed permission is Edit-Write on the "other" folder. | The server of both users is updated, if the other folder is not shared with permissions, you can't the other foder | :heavy_check_mark: | |
9 | Move a shared folder with permissions | 1. Move a shared folder with permissions inside another folder from Desktop 2. Create a file inside the sahred folder| The server of both users is updated | :heavy_check_mark:  | |
10 | Move a shared folder with permissions | 1. Move a shared folder with permissions inside another folder from Desktop 2. Create a file inside the shared folder 3. Update the file from user1| The Desktop update the file | :heavy_check_mark:  | |
11 | Share a file with many users | 1. Having one file shared with 80 users | The Desktop works fine, list of users can be scrolled | :heavy_check_mark:  | tested with 25 |
12 | Share a Folder with Password protect| 1. Create a new folder 2. Share with oC 3. Check Share link option 4. Check Password protection 5. Introduce the password 6. Press Enter or Click on Set password option | You can write the password| :heavy_minus_sign:  | no password option exists -- not a bug |
13 | Share a Folder with Set expiration date| 1. Create a new folder 2. Share with oC 3. Check Share link option 4. Check the Set expiration date 5. Introduce one day for expiration date | You can configurate the expiration date | :no_entry: | -> #8937 |
14 | Share a Folder with Allow editing| Create a new folder 2. Share with oC 3. Check Share link option 4. Check Allow editing option  | The Folder can allow editing| :heavy_check_mark:  | |
15 | Share a File with Password protect| 1. Create a new file 2. Share with oC 3. Check Share link option 4. Check Password protection 5. Introduce the password 6. Press Enter or Click on Set password option  | You can write the Password protect | :heavy_minus_sign:  | no password option exists -- not a bug | |
16 | Share a File with Set expiration date| 1. Create a new file 2. Share with oC 3. Check Share link option 4. Check the Set expiration date 5. Introduce one day for expiration date |You can configurate the expiration date | :no_entry:  | -> #8937  |
17 | Modify the Set expiration date option on the server| 1. Create a new file on the Desktop 2. Share with oC 3. Create Public link at the client; 4. Check the Set expiration date 5. Introduce one day for expiration date 6. Go to the Server and change the date 7. Go to the Desktop 8. Wait to sync | You can see the new date on Set expiration date | :heavy_check_mark: : | |
18 | Public link a file and download it | 1) right click a file - open the share menu; 2) Create and copy public link to clipboard; 3) paste link in a web browser; 4) download | The file can be downloaded | :heavy_check_mark:  |
19 | Public link a folder and download it | 1) right click a folder - open the share menu; 2) Create public link; 3) 3dots menu -> "copy public link to clipboard (direct download)"; 4) paste link in a web browser | A zip archive of the folder can be downloaded | :heavy_check_mark:  |
20 | Public link to a file and disable it after | use the sharing menu to delete the public link | The link reports File not found. | :heavy_check_mark: |
21 | Public link to a file and 'add to your owncloud' a user that belongs to the server | | The file is shared | :heavy_check_mark:  |
22 | ~Public link to a file and 'add to your owncloud' a user that belongs to the server and has been already shared another file~ | | The file is shared | :heavy_minus_sign:  | That only tests the server
23 | Public link to a file and 'add to your owncloud' with a user that belongs to another server | | The file is shared | :heavy_check_mark:  |
24 | Public link to a file and 'add to your owncloud' with an LDAP user | | The file is shared | :heavy_check_mark:  |
25 | ~Optionally Public link to a file and 'add to your owncloud' with an Active Directory user~ | | The file is shared | :heavy_check_mark:   |
26 | ~Public link to a file and 'add to your owncloud' with a Shibboleth user autoprovisioned~ | | The file is shared | :heavy_minus_sign:   | Shiboleth is not supported
27 | Try to reshare a file originally received from an LDAP user | | The file can be reshared, if "Allow Share" was set, otherwise it cannot be reshared | :heavy_check_mark:   | -> #8941
28 | Public link to a file and 'add to your owncloud' with a user that belongs to another oC server below 8.2.X version | `docker run --rm -ti -p 8181:80 owncloud:8.1` | The file is added to the old server | :heavy_check_mark:  |
29 | Public link to a file and set a password with special characters | | The file is shared with a pwd | :heavy_check_mark:  |
30 | Public link to a file and set a password. Modify Password Policies and retry | 1) enable password policy app allow passwords with 4 chars; 2) share with a 4 char password; 3) change password policy to min 8 chars. 4) use client sharing dialog to remove the password, and re-enter the same; 5) enter a password with 8 chars  | An error appears stating min passwiord length 8 for the short password; a longer password works  | :heavy_check_mark:   |

**Comment by swoichha** [comment]
#### 9.2 Share with Users and Groups

TestID | Test Case | Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------
1 | ~~Enable encryption app and encryption default module~~ | Encryption is enabled correctly | :construction: |
2 | Share a file with a user that has been already shared another file | Both files appear in server | ✔️ |
3 | Share a folder with a user | The folder appears in server | ✔️ |
4 | Share a folder with a user that has been already shared another folder and another file | Both folders appear and the file in server | ✔️ |
5 | Share a file with a group| The file appears in server | ✔️ | 
6 | Share a folder with an LDAP user | The folder appears in server | :heavy_check_mark:  |
7 | ~Share a file with an Active Directory user~ | The file appears in server | :heavy_minus_sign:  |
8 | Share a folder with a Shibboleth user autoprovisioned| The folder appears in server | :heavy_minus_sign:  | SKIP. We don't support Shiboleth
8a | Share a file with a Shibboleth SSO user| The file appears in server | :heavy_minus_sign:   | SKIP: We don't support Shiboleth
9 | Share a folder with remote user on a 10.2.1 server| The folder appears | ✔️ |
10 | Receive a federated share from a 10.2.1 server | When auto-accept is disabled: Notification with Dismiss/Accept/decline appears in the "Server Activity" tab | :heavy_check_mark:   |
11 | Share a file with one user whose name contains special characters | The file appears in server | ✔️ |
12 | Share a folder with one users a file whose name contains special characters | The folder appears in server | ✔️ |
13 | Share a file bigger than 1 GB with one user | The file appears in server | ✔️ |
14 | Try to federate share a file with one user | The file appears in server, expect delays | ✔️ |
15 | Share a file from SFTP with a user | The file appears in server | ✔️ |
16 | ~Share a folder from SMB with a user~ | The folder appears in server | :construction:  |
17 | Share a file from WND with a user | The file appears in server | ✔️ |
18 | ~Share a folder from Sharepoint with a user~ | The folder appears in server | :construction:  |
19 | Share a file from a Federated Sharing with a user | The file appears in server, expect delays | ✔️ |
20 | ~Share a folder from redirect server with a user~ | The folder appears in server | :construction: | @michaelstingl ???
21 | ~Using a redirect server, Share a file with a user~ | The file appears in server | :construction: |
22 | Share a file with user B with Can Share permissions enabled. Login as User B and reshare it | The file can be reshared in the server with other user| ✔️ |
23 | Share a file with user B with Can Edit permissions enabled. Login as User B and edit it  | The file can be edited in the server with other user | ✔️ |
24 | Share a file with user B with Change permission enabled. Login as User B and change it  | The file can be changed in the server with other user | ✔️ |
25 | Share a file with user B with Can Edit permissions enabled. Login as User B and edit it. Change via Desktop the permissions. try to edit it again | The file can be edited in the server with other user | ✔️ ❓ |
26 | Share a folder with user B with Can Share permissions enabled. Login as User B and reshare it | The folder can be reshared in the server with other user | ✔️ ❓ | Should be after 22 in test plan.
27 | Share a folder with user B with Can Edit permissions enabled. Login as User B and edit it  | The folder can be edited in the server with other user | ✔️ |
28 | Share a folder with user B with Change permission enabled. Login as User B and change it  | The folder can be changed in the server with other user | ✔️ |
29 | Share a folder with user B with Create permissions enabled. Login as User B and create a file in it | The folder can be edited creating a file in the server with other user | ✔️ |
30 | Share a folder with user B with Create permissions enabled. Login as User B and create a folder in it | The folder can be edited creating a folder in the server with other user | ✔️ |
31 | Share a folder with user B with Can Edit permissions enabled. Login as User B and delete it  | The folder can be edited in the server with other user | ✔️ |
32 | Share a folder with user B with Can Edit permissions enabled. Login as User B and edit it. Change via Desktop the permissions. Try to edit it again | The folder cannot be edited the second time |  ✔️ |
33 | Share a file with user B without Can Share permission enabled. Login as User B and try to reshare it | The file cannot be reshared |  :heavy_check_mark: |
34 | Share a file with user B without Can Edit permission enabled. Login as User B and try to edit it  | The file cannot be edited | :heavy_check_mark: |
35 | Share a file with user B without Change permission enabled. Login as User B and try to change it  | The file cannot be changed | :heavy_check_mark: |
36 | Share a folder with user B without Can Share permission enabled. Login as User B and try to reshare it | The folder cannot be reshared | :heavy_check_mark:|
37 | Share a folder with user B without Can Edit permissions enabled. Login as User B and try to edit it  | The folder cannot be edited | :heavy_check_mark:  |
38 | Share a folder with user B without Create permissions enabled. Login as User B and try to create a file in it  | The folder cannot be edited creating a file in it | :heavy_check_mark: |
39 | Share a folder with user B without Change permissions enabled. Login as User B and try to change it  | The folder cannot be changed | :heavy_check_mark:|
40 | Share a folder with user B without Delete permissions enabled. Login as User B and try to delete it  | The folder cannot be edited deleting files/folders in it | :heavy_check_mark: |
41 |~With Password Policy App enabled, Share link a file with a user  and set a password that matches with the pwd policy~| The file is shared with pwd |:construction:|
42 | ~With Password Policy App enabled, Share link a file with a user  and set a password that does not match with the pwd policy~| An error should be shown | :construction:| 
43 | Share link a file with a user and set a password and a expiration date | The file is shared with pwd and expiration date |:heavy_check_mark:  |
44 | Share link a file with a user and set a password and a expiration date. Change the expiration date | The file is shared with the updated expiration date | :heavy_check_mark:  |
45 | Share link a file with a user and set a password and a expiration date. Change the pwd | The file is shared with the updated pwd | :heavy_check_mark:  |
46 | Try to search a user that has already been shared the file in the users searchbox | The user should not be shown | :heavy_check_mark: |
47 | Try to search yourself in the users searchbox | The user should not be shown | :heavy_check_mark:|
48 | Search with minimum characters required. For example: 'pat' | All users that contains this pattern should be shown |:heavy_check_mark:  |
49 | Try to search a remote user in the users searchbox | The user should be shown | :heavy_check_mark: | 
50 | Share a file with more than 4 users | The users should be listed and ordered chronologicaly. If is needed a scrollbar will appear | :heavy_check_mark: |
51 | Sharing API  | The users should be listed and ordered chronologicaly. If is needed a scrollbar will appear | :heavy_check_mark:  |
52 | Enforce password protection  | Password should be always required | :heavy_check_mark:   |
53 | Enforce password protection enabled. Try to disable the pwd  | Password should be always required |:heavy_check_mark: | 
54 | Set default expiration date  | The expiration date should be set by default |  :heavy_check_mark:   |
55 | Allow resharing disabled and Can Share enabled  | The user cannot reshare files | :heavy_check_mark:  |
56 | Restrict users to only share with users in their groups | It should appear only the users from the group |:heavy_check_mark:| 
57 | Disallow username autocompletion in share dialogs | Full username needs to be entered in order to sharing | :heavy_check_mark:  |

**Comment by swoichha** [comment]
### 10. Selective_Sync

TestID | Test Case | Steps to reprouce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 | Check that in the setup options after fill in the server and the user you have to choose the server and local folder to sync| 1. Start to setupo oC Desktop 2. Fill in server, user and password 3. The server and local folder to sync is shown|1. On server (you have to choose what do yo want to sync) 2. local folder (the local folder where the data are sync) |:heavy_check_mark: |  |
2a | Linux+Mac: Check that in the setup options the checkbox is enable to sync all the account by default | 1. Start to setup oC Desktop 2. Fill in server, user and password 3. Click on "Choose what to sync" | At window is opened with the directory tree |:heavy_check_mark: | Done for Ubuntu 20.04.2 LTS  |
2b | Win10: Check that in the setup options the checkbox is enable to "use virtual files" default | 1. Start to setup oC Desktop 2. Fill in server, user and password 3. Click on "Synchronize everything from server"; 4. "Choose what to sync" | At window is opened with the directory tree |:heavy_check_mark: |  |
3 | Check that all files and folders are synced when you select to sync all | 1. Start to setup oC Desktop 2. Fill in server, user and pass 3. Click on "Sync everything from server" and wait until everything is sync| Everything is synced |:heavy_check_mark: |  |
4 | Verify that you can select to sync only one folder | 1. Start to setup the oC Desktop 2. The new window select what folder sync is shown 3. Select to sync only one folder 4. Click on OK | Only one folder is sync |:heavy_check_mark: |  |
5 | Verify that the folder to select are sorted by name/size| 1. Start to setup the oC Desktop 2. The new window to select what folder sync is shown 3. Select all the folder except one of them| Everything except one folder is synced  |:heavy_check_mark: |  |
6 | Check that there are another option: skip sync, so that nothing is synced | 1. Start to setup oC Desktop 2. Fill in server, user and password 3. Click on "Manually create folder sync connection"| The settings windows is openend but no local folder is configured|:heavy_check_mark: |  |

#### 10.1 Choose what to sync from the setup window

TestID | Test Case | Steps to reprouce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 | Verify that the folders that you unselect are not sync in Desktop| 1. Start to setupo the oC Desktop 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolders | Check that the folder unselected are not sync|:heavy_check_mark: |  |
2 | The folder without subfolder doesn't show the arrow| 1. Start to setup the oC Desktop app 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolder | When you click on a folder without subfolder the arrows disappear sometimes|:x: |  The arrows are still there for folder without subfolder https://github.com/owncloud/client/issues/8935|
3 | Verify that the folders that you unselect are not sync in Desktop| 1. Start to setupo the oC Desktop 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolders | The parent folder is sync but not the subfolder|:heavy_check_mark: |  |
4 | Upload from the server some files in a folder which is sync| 1. Start to setup the oC Desktop app 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolders 4. Upload files from server | The files are download in the desktop client|:heavy_check_mark: |  |
5 | Upload from the Desktop some files | 1. Start to setup the oC Desktop 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolders 4. Upload files from server in a folder wich is not sync | The files are sync with the server|:heavy_check_mark: |  |
6 | Upload from the server some files in a folder which is  not sync | 1. Start to setupo the oC Desktop 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolder 4. Upload files from server which is not sync | The files are not downloads in the Desktop client |:heavy_check_mark: |  |
7 | Delete a folder sync, it is removed from the Desktop | 1. Start to setup the oC Desktop 2. The new window to select what folder sync is shown 3. Unselect different kids of folder with and witout subfolders 4. Delete a folder sync, it is removed from the desktop | The folder is removed int he server |:heavy_check_mark: |  |
8 | Delete a sync folder from the server | 1. Start to setup th oC Desktop 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolders 4. Delete a sync folder from the server | The folder is removed int he Desktop |:heavy_check_mark: |  |
9 | Create a folder in the root directory from the server | 1. Start to setup th oC Desktop 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolders 4. Create a folder in the root directory from the server | The folder is sync |:heavy_check_mark: |  |
10 | Rename a folder that is sync | 1. Start to setup th oC Desktop 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolders 4. Rename a folder that is sync | It is renamed in the window "Choose what to sync" |:heavy_check_mark: |  |
11 | Delete a folder that is sync | 1. Start to setup th oC Desktop 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolders | It is not shown in the window "Choose what to sync" |:heavy_check_mark: |  |
12 | Verify that when yo unselect all the folder, the root files continue sync| 1. Start to setup th oC Desktop 2. The new window to select what folder sync is shown 3. Unselect different kinds of folder with and without subfolders | Root files are sync |:heavy_check_mark: |  |
13 | When you click on "Choose what to sync" folder tree is shown | 1. Click on "Choose what to sync" | It takes folder time to appear the folder tree,  |:heavy_check_mark: |  |

**Comment by swoichha** [comment]
#### 10.2 "Skip folders configuration"

TestID | Test Case | Steps to reprouce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 |Skip folder configuration| 1. Start to setup oC Desktop 2. Fill in server, user and password 3. Click on "Skip folders configuration" | 1. No one local folder is sync 2. The setting window is opened and the account is registered| :heavy_check_mark:  | There is no skip button. The only way to reproduce this case is to mark "Manually create folder sync connection" on and click the "connect" button |
2 |Sync only one folder the settings view | 1. Start to setupo oC Desktop 2. Fill in server, user and password 3. Click on "Skip folders configuration" 4. Click on Settings window/add folder| The folder start to sync and the folder is available from the tray icon| :heavy_check_mark:  |  |

#### 10.3 "Add Folder..."

TestID | Test Case | Steps to reprouce| Expected Result | Result | Related Comment
------------ | ------------- | -------------- | ----- | ------ | ------
1 |Skip folder configuration from the setup window and select the local and remote folder front he setting window| 1. Start to setup oC Desktop 2. Fill in server, user and password 3. Click on "Skip folders configuration" 4. Click on Add folder 5. Select the local folder (choose an existing one) 6. Select the remote folder, click on continue 7. Select the subfolders of the remote folder and click on add | 1. The local folder selected is sync wich the remote one| :heavy_check_mark:  |  |
2 |Select a remote folder with special characters| 1. Start to setup oC Desktop 2. Fill in server, user and password 3. Click on "Skip folders configuration" 4. Click on Add folder 5. Select the local folder (create a new folder with special characters) 6. Select the remote folder with special characters, click on continue 7. Select the subfolders of the remote folder and click on add | 1. The local folder selected is sync wich the remote one| :heavy_check_mark:  |  |

**Comment by individual-it** [comment]
blocked waiting for reviews of @kulmann @pascalwengerter

**Comment by Talank** [comment]
File creation(https://github.com/owncloud/client/issues/8971#issuecomment-909884348) is currently blocked, until https://github.com/owncloud/client/issues/9281#issuecomment-1002033359 is fixed.

**Comment by Talank** [comment]
Now that we have testplan 2.10, which is updated version of this test plan, I think we should wrap up with opened PRs and then open a new issue with the `not yet automated tests from testplan 2.10`. And let's not create a new PR referencing this issue. What do you think @saw-jan @individual-it

**Comment by individual-it** [comment]
@Talank sounds good, let's close this one and track the remaining work in a new issue

**Comment by kiranparajuli589** [comment]
the new issues will be created as per the discussions in the above comments.
cc @Talank @saw-jan
